addons/auth_signup/controllers/main.py

```diff
@@ -20,11 +20,19 @@ def web_login(self, *args, **kw):
         ensure_db()
         response = super().web_login(*args, **kw)
         response.qcontext.update(self.get_auth_signup_config())
-        if request.httprequest.method == 'GET' and request.session.uid and request.params.get('redirect'):
-            # Redirect if already logged in and redirect param is present
-            return request.redirect(request.params.get('redirect'))
+        if request.session.uid:
+            if request.httprequest.method == 'GET' and request.params.get('redirect'):
+                # Redirect if already logged in and redirect param is present
+                return request.redirect(request.params.get('redirect'))
+            # Add message for non-internal user account without redirect if account was just created
+            if response.location == '/web/login_successful' and kw.get('confirm_password'):
+                return request.redirect_query('/web/login_successful', query={'account_created': True})
         return response
 
+    @http.route()
+    def login_successful_external_user(self, account_created=False):
+        return request.render('auth_signup.login_successful', {'account_created': account_created})
```

Comment:
Instead of changing the rendered template, idea would be to add / propagate the parameter and still render on web.login_successful (with its inherited template like every qweb template).